default.nix

```
@@ -1 +1,122 @@
-import ./lib/overlay.nix (_: import ./packages.nix)
+{ system ? builtins.currentSystem, flakeLock ? import ./flake.lock.nix
+, nixpkgs ? import flakeLock.nixpkgs {
+  inherit system;
+  config = { };
+  overlays = [ ];
+}, lib ? nixpkgs.lib, rust-analyzer-src ? flakeLock.rust-analyzer-src }:
+with builtins;
+with lib;
```

This should be able to replace `flake.lock.nix`, also I don't think flakeLock should be an argument here
```suggestion
with builtins;

let
  getFlake = name:
    with (fromJSON (readFile ./flake.lock)).nodes.${name}.locked;
    fetchTarball {
      url = "https://github.com/${owner}/${repo}/archive/${rev}.tar.gz";
      sha256 = narHash;
    };

in { system ? currentSystem
, nixpkgs ? import (getFlake "nixpkgs") { inherit system; }, lib ? nixpkgs.lib
, rust-analyzer-src ? getFlake "rust-analyzer-src" }:

with lib;

```